ecommerce/views_test.py

```diff
@@ -612,22 +615,28 @@ def test_patch_basket_invalid_run(basket_client, basket_and_coupons, is_program)
 
     # If the product is a course, create a new run on a different course which is invalid.
     # If the product is a program, create a new run on a different program.
-    other_run = (
-        CourseRunFactory.create()
-        if is_program
-        else CourseRunFactory.create(
-            course__program=product.content_object.course.program
-        )
+    other_run_id = (
+        (
+            CourseRunFactory.create()
+            if is_program
+            else CourseRunFactory.create(
+                course__program=product.content_object.course.program
+            )
+        ).id
+        if is_selected
+        else None
```

Comment:
This might be a bit cleaner:

```python
    course_run_params = (
        dict(course__program=product.content_object.course.program)
        if is_program
        else {}
    )
    other_course_run = CourseRunFactory.create(course_run_params)
    other_run_id = course_run.id if is_selected else None
```

Reply:
done